src/EnergyPlus/PipeHeatTransfer.cc

```diff
@@ -1748,24 +1747,23 @@ Real64 PipeHTData::CalcPipeHeatTransCoef(EnergyPlusData &state,
     LoopNum = this->plantLoc.loopNum;
 
     // since the fluid properties routine doesn't have Prandtl, we'll just use water values
-    idx = 0;
+    int idx = 0;
     while (idx < NumOfPropDivisions) {
         if (Temperature < Temps[idx]) {
-            if (idx == 1) {
-                PRactual = Pr[idx];
-            } else if (idx > NumOfPropDivisions) {
-                PRactual = Pr[NumOfPropDivisions - 1];
-            } else {
-                InterpFrac = (Temperature - Temps[idx - 1]) / (Temps[idx] - Temps[idx - 1]);
-                PRactual = Pr[idx - 1] + InterpFrac * (Pr[idx] - Pr[idx - 1]);
-            }
-            break; // DO loop
-        } else {
-            PRactual = Pr[NumOfPropDivisions - 1];
+            break;
         }
         ++idx;
     }
 
+    if (idx == 0) {
+        PRactual = Pr[idx];
+    } else if (idx >= NumOfPropDivisions) {
+        PRactual = Pr[NumOfPropDivisions - 1];
+    } else {
+        InterpFrac = (Temperature - Temps[idx - 1]) / (Temps[idx] - Temps[idx - 1]);
+        PRactual = Pr[idx - 1] + InterpFrac * (Pr[idx] - Pr[idx - 1]);
+    }
+
```

Comment:
Fix is here. This was an off-by-one error, introduced when modernizing from Array1D to constexpr std::array